### PR TITLE
Feedback on goal

### PR DESCRIPTION
**Pull request by emilyb7**
created components for rating goal and giving feedback on rating

### CODE REVIEW

File: tests/reducer-test.js [hunk 1]
```diff
@@ -89,7 +93,7 @@ tape('test reducer case SAVE_NEW_GOAL: adds new goal object to goals array and c
 
 tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t) => {
 
-  const myGoal = { name: 'this is my goal', };
+  const myGoal = { name: 'this is my goal'};
```
Comment (des-des) [inline tests/reducer-test.js]: inconsistent spacing...eoin

File: tests/reducer-test.js [hunk 2]
```diff
@@ -98,8 +102,124 @@ tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t)
...
 });
+
+tape('test reducer step_rate_goal: step and previousStep changed', (t) => {
+
+  const initialState = defaultState;
+  const actionStepRateGoal = {
+    type: types.STEP_RATE_GOAL,
+  };
+
+  t.equal(reducer(initialState, actionStepRateGoal).step, steps.RATE_GOAL, "step rate goal sets correct step");
```
Comment (des-des) [inline tests/reducer-test.js]: too long!

File: tests/reducer-test.js [hunk 3]
```diff
@@ -98,8 +102,124 @@ tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t)
...
+  t.equal(reducer(initialState, actionStepRateGoal).previousStep, steps.VIEW_GOAL, "step add goal sets correct step");
```
Comment (des-des) [inline tests/reducer-test.js]: You only need to call the reducer once.

ie
```js
const nextState = reducer(initialState, actionStepRateGoal);
t.equal(nextState.step, steps.RATE_GOAL, "step rate goal sets correct step");
t.equal(nextState.previousStep, steps.VIEW_GOAL, "step add goal sets correct step");
```
or better again
```js
const nextState = reducer(initialState, actionStepRateGoal);
t.equal(
  nextState.step,
  steps.RATE_GOAL,
  "step rate goal sets correct step"
);
t.equal(
  nextState.previousStep,
  steps.VIEW_GOAL,
  "step add goal sets correct step"
);
```

File: tests/reducer-test.js [hunk 4]
```diff
@@ -98,8 +102,124 @@ tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t)
...
+
+tape('test reducer step_feedback: step and previousStep changed', (t) => {
+
+  const initialState = defaultState;
+  const actionStepFeedback = {
+    type: types.STEP_FEEDBACK,
+  };
+
+  t.equal(reducer(initialState, actionStepFeedback).step, steps.FEEDBACK, "step feedback sets correct step");
+  t.equal(reducer(initialState, actionStepFeedback).previousStep, steps.RATE_GOAL, "step feedback sets correct step");
```
Comment (des-des) [inline tests/reducer-test.js]: again, just call the reducer once!

File: tests/reducer-test.js [hunk 5]
```diff
@@ -98,8 +102,124 @@ tape('test reducer case SELECT_GOAL: current goal is set to selected goal', (t)
...
+tape('test reducer RESET_UPDATE_COUNT: updateCount and syncDBCount set to 0', (t) => {
+
+  const initialState = {
+    goals: [{id: 1, updateCount: 1, syncDBCount: 2}]
+  };
+  const newState = {
+    goals: [{id: 1, updateCount:0, syncDBCount: 0}]
+  };
+  const actionResetUpdateCount = {
+    type: types.RESET_UPDATE_COUNT,
```
Comment (des-des) [inline tests/reducer-test.js]: Why do you want to do this? Surely you update count should always increase, if it goes down then bad things can happen?

Comment (emilyb7) [inline tests/reducer-test.js]: permission from eoin to ignore this

File: tests/reducer-test.js (hunk 1 above)
Comment (emilyb7) [inline tests/reducer-test.js]: useful feedback @JWLD!

File: app/reducers/index.js [hunk 6]
```diff
@@ -46,7 +41,73 @@ export default (state = defaultState, action) => {
         ...state,
         step: steps.VIEW_GOAL,
         previousStep: steps.GOALS_LIST,
-        currentGoal: action.goal,
+        currentGoal: {
+          ...action.goal,
+          newRating: {},
+        }
+      }
+    case 'STEP_RATE_GOAL':
```
Comment (emilyb7) [inline app/reducers/index.js]: fixed on brach...

File: app/reducers/index.js [hunk 7]
```diff
@@ -46,7 +41,73 @@ export default (state = defaultState, action) => {
...
+        return {
+          ...state,
+          goals: state.goals.map((goal) => {
+            return action.id === goal.id ? { ...goal, pendingSync: {open: false} } : goal;
+          })
+        }
+    case 'RESET_UPDATE_COUNT':
+      return {
+        ...state,
+        goals: state.goals.map((goal) => {
```
Comment (emilyb7) [inline app/reducers/index.js]: done

File: tests/reducer-test.js (hunk 1 above)
Comment (emilyb7) [inline tests/reducer-test.js]: fixed

File: app/containers/main-container.jsx [hunk 8]
```diff
@@ -36,6 +39,10 @@ const mapDispatchToProps = dispatch => ({
   onSelectAvatar: (avatar) => { dispatch(actionsAddNewGoal.selectAvatar(avatar)) },
   saveNewGoal: (goal) => { dispatch(actionsAddNewGoal.saveNewGoal(goal)) },
   onSelectGoal: (goal) => { dispatch(actionsGoalsList.selectGoal(goal)) },
+  stepRateGoal: () => { dispatch(actionsViewGoal.stepRateGoal())},
+  onMoveSlider: (rating) => { dispatch(actionsRateGoal.moveSlider(rating))},
+  stepFeedback: () => { dispatch(actionsRateGoal.stepFeedback())},
+  onInputFeedback: (input) => {dispatch(actionsFeedback.inputFeedback(input))},
 });
 
 export default connect(mapStateToProps, mapDispatchToProps)(MainContent);
```
Comment (emilyb7) [inline app/containers/main-container.jsx]: @des-des don't get this! do you have another example?

**Pull request merged**